### PR TITLE
[Table/EditableCell] Only update saved and dirty values if value changed

### LINKED ISSUE #1261: [Table] Editing body cell too quickly after focus is buggy
(Note: This is happening on a branch, so maybe it's due to something I did locally. Capturing anyway.)

### Bug report

- __Package version(s)__: v1.20.0
- __Browser and OS versions__: Chrome 59 on OS X Sierra

#### Steps to reproduce

1. Double click an editable body cell in the table.
1. Immediately start typing. 

#### Actual behavior

Your typing is partially or fully replaced by the original cell value. However, if you wait a moment before typing, everything is fine.

edit-bug

#### Expected behavior

Your typing is honored no matter how soon you do it after focus.

### PR DESCRIPTION
**Pull request by gscshoyru**
#### Changes proposed in this pull request:

We were updating saved and dirty value to match the value whenever _any_ prop changed, no matter what. Even if `shouldComponentUpdate` would have returned false, because `componentWillReceiveProps` is always true. So if there were any changes made before rendering got there (because batched rendering), it'd reset those changes. Now this no longer happens. 

#### Reviewers should focus on:

Is `componentWillReceiveProps` even the right place for this? It apparently can't live in `componentWillUpdate` because that won't take a `setState`, but... is there a better lifecycle spot for this? 

Or more generally: is there a better way to do what I'm doing?

### CODE REVIEW

File: packages/table/src/cell/editableCell.tsx [hunk 1]
```diff
@@ -72,7 +72,9 @@ export class EditableCell extends React.Component<IEditableCellProps, IEditableC
 
     public componentWillReceiveProps(nextProps: IEditableCellProps) {
         const { value } = nextProps;
-        this.setState({ savedValue: value, dirtyValue: value });
+        if (value !== this.props.value) {
+            this.setState({ savedValue: value, dirtyValue: value });
+        }
```
Comment (cmslewis) [inline packages/table/src/cell/editableCell.tsx]: Cool! Now, how do we unit-test this?

Comment (gscshoyru) [reply packages/table/src/cell/editableCell.tsx]: Heh. It should be doable, I guess.

File: packages/table/test/editableCellTests.tsx [hunk 2]
```diff
@@ -72,6 +72,37 @@ describe("<EditableCell>", () => {
...
+            <EditableCell value="test-value-5000" onCancel={onCancel} onChange={onChange} onConfirm={onConfirm} />,
+        );
+
+        // double click to edit
+        doubleClickToEdit(elem);
+        const input = getInput(elem);
+        expect(input.element).to.equal(document.activeElement);
+
+        // edit
+        input.change("my-changed-value");
```
Comment (cmslewis) [inline packages/table/test/editableCellTests.tsx]: Whaaat is this glorious `.change(...)` magic.

File: packages/table/test/editableCellTests.tsx [hunk 3]
```diff
@@ -72,6 +72,37 @@ describe("<EditableCell>", () => {
...
+        expect(onChange.called).to.be.true;
+        expect(onCancel.called).to.be.false;
+        expect(onConfirm.called).to.be.false;
+        expect(elem.find(".pt-editable-content").text()).to.equal("my-changed-value");
+
+        harness.mount(
```
Comment (cmslewis) [inline packages/table/test/editableCellTests.tsx]: So this is functionally equivalent to `setProps`?

**Pull request merged**